materialize-postgres/client.go

```diff
@@ -112,15 +112,32 @@ func (c *client) CreateTable(ctx context.Context, tc sql.TableCreate) error {
 		}
 	}
 
-	statements := []string{}
 	if res.AdditionalSql != "" {
-		statements = append(statements, txnStatements(tc.TableCreateSql, res.AdditionalSql))
+		txn, err := c.db.BeginTx(ctx, nil)
+		if err != nil {
+			return fmt.Errorf("db.BeginTx: %w", err)
+		}
+		defer txn.Rollback()
+
+		if _, err := txn.ExecContext(ctx, tc.TableCreateSql); err != nil {
+			return fmt.Errorf("executing CREATE TABLE statement: %w", err)
+		} else if _, err := txn.ExecContext(ctx, res.AdditionalSql); err != nil {
+			return fmt.Errorf("executing additional SQL statement '%s': %w", res.AdditionalSql, err)
+		} else if err := txn.Commit(); err != nil {
+			return fmt.Errorf("committing transaction: %w", err)
+		}
+
+		log.WithFields(log.Fields{
+			"table": tc.Identifier,
+			"query": res.AdditionalSql,
+		}).Info("executed AdditionalSql")
 	} else {
-		statements = append(statements, tc.TableCreateSql)
+		if _, err := c.db.ExecContext(ctx, tc.TableCreateSql); err != nil {
```

Comment:
TableCreateSql can still include multiple statements (such as COMMENTs), and should be within a transaction right?

Reply:
Ah yeah that's probably a good call. Updated the code.